src/views/Search.vue

```diff
@@ -0,0 +1,151 @@
+    div Start: {{start.format()}}
+    div End: {{stop.format()}}
+    //div
+      label Use regex
+      input(type="checkbox" v-model="use_regex")
+    div
+      label Filter AFK
+      input(type="checkbox" v-model="filter_afk")
+    //div.form-row
+      div.form-group.col-md-6
+        | Start
+        input.form-control(type="date", :max="today", v-model="startdate")
+      div.form-group.col-md-6
+        | End
+        input.form-control(type="date", :max="tomorrow", v-model="enddate")
+
+    div.form-inline
+
+  div(v-if="status == 'searching'")
+    div #[icon(name="spinner" pulse)] Searching...
+
+  div(v-if="events != null")
+    hr
+
+    div.form-group
+      select.form-control(v-model="vis_method")
+        option(value="eventlist") Event List
+        option(value="timeline") Timeline
+        option(value="summary") Summary
+        option(value="raw") Raw JSON
+
+    div(v-if="vis_method == 'timeline'")
+      aw-timeline(type="simple", :event_type="event_type", :events="events")
+    div(v-if="vis_method == 'eventlist'")
+      aw-eventlist(:events="events")
+    div(v-if="vis_method == 'summary'")
+      input.form-control(type="text" v-model.lazy.trim="summaryKey" placeholder="data key" style="margin-bottom: 1em;")
+      aw-summary(:fields="events", :colorfunc="colorfunc", :namefunc="namefunc")
+    div(v-if="vis_method == 'raw'")
+      pre {{ events }}
+
+    div
+      | Didn't find what you were looking for?
+      br
+      | Add a week to the search: #[b-button(size="sm" variant="outline-dark" @click="start = start.subtract(1, 'week'); search()") +1 week]
+
+</template>
+
+<style scoped lang="scss"></style>
+
+<script>
+import _ from 'lodash';
+import moment from 'moment';
+import { canonicalEvents } from '~/queries';
+
+import 'vue-awesome/icons/search';
+import 'vue-awesome/icons/spinner';
+import 'vue-awesome/icons/angle-double-down';
+import 'vue-awesome/icons/angle-double-up';
+
+export default {
+  name: 'Search',
+  data() {
+    return {
+      pattern: '',
+      vis_method: 'eventlist',
+      event_type: 'currentwindow',
+      events: null,
+      error: '',
+      status: null,
+      hostname: '',
+
+      // Options
+      show_options: false,
+      use_regex: true,
+      filter_afk: true,
+      start: moment().subtract(1, 'day'),
+      stop: moment().add(1, 'day'),
+
+      /* Summary props */
+      summaryKey: 'title',
+      colorfunc: null,
+      namefunc: null,
+    };
+  },
+  mounted: async function () {
+    this.colorfunc = this.summaryKeyFunc;
+    this.namefunc = this.summaryKeyFunc;
+    await this.$store.dispatch('buckets/ensureBuckets');
+    this.hostname = Object.keys(this.$store.getters['buckets/bucketsByHostname'])[0];
+  },
+  methods: {
+    search: async function () {
+      let query = canonicalEvents({
+        bid_window: 'aw-watcher-window_' + this.hostname,
+        bid_afk: 'aw-watcher-afk_' + this.hostname,
+        filter_afk: this.filter_afk,
```

Comment:
I don't think a "filter-afk" option is a good idea (assuming that we want the ability to delete events in the results).

Let's say someone searches for an event and wants to delete it, if that event spans both before and after and AFK intersection the user will delete the part both before and after that period which was possibly not intended.

So if we have a view somewhere where it's possible to delete event, that view should never be transformed. Adding more info to it such as with "split_url" or "categorize" is fine, but changing or removing keys/vals in the data or changing duration/timestamp on an event is not OK. And canonicalEvents does both flooding and filter_period_intersect so that's not good.

Reply:
Yeah, I thought a bit about this but didn't know of a reasonable way to make it work with the upcoming merge of web-events (which was more or less point of refactoring into `canonicalEvents`).

I really don't want the user to have to think "which bucket could the thing I'm looking for be in?".

The simple solution here is to simply not let the user modify events in search results. If they want to delete/censor events there are much better ways than manually searching through them and deleting them one by one.

I'll make an example in aw-client that allows the user to censor/delete events.